File: LondonDataServices.IDecide.Core/Models/Foundations/DecisionTypes/DecisionType.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Collections.Generic;
using LondonDataServices.IDecide.Core.Models.Foundations.Decisions;
using Microsoft.AspNetCore.Mvc.ModelBinding;

namespace LondonDataServices.IDecide.Core.Models.Foundations.DecisionTypes
{
    public class DecisionType
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }

        [BindNever]
        public List<Decision> Decisions { get; set; } = new List<Decision>();
    }
}

File: LondonDataServices.IDecide.Portal.Client/src/models/decisionTypes/decisionType.ts
import { PatientDecision } from '../patientDecisions/patientDecision';

export class DecisionType {
    public id?: string;
    public name?: string;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public decisions?: PatientDecision[];

    constructor(type?: Partial<DecisionType>) {
        this.id = type?.id || '';
        this.name = type?.name || '';
        this.createdBy = type?.createdBy || '';
        this.createdDate = type?.createdDate ? new Date(type.createdDate) : undefined;
        this.updatedBy = type?.updatedBy || '';
        this.updatedDate = type?.updatedDate ? new Date(type.updatedDate) : undefined;
        this.decisions = type?.decisions || [];
    }
}